aws/rust-runtime/aws-config/src/sts/util.rs

```diff
@@ -39,3 +44,20 @@ pub(crate) fn default_session_name(base: &str, ts: SystemTime) -> String {
     let now = ts.duration_since(UNIX_EPOCH).expect("post epoch");
     format!("{}-{}", base, now.as_millis())
 }
+
+// A subset of functionality extracted from `endpoint_lib::arn::Arn::parse`.
+// `Arn` is `pub(crate)` within generated SDKs, making it inaccessible from `aws-config`.
+// As a result, a subset is inlined here, with less defensive verification
+// since it only deals with the string-form ARN returned by STS.
+fn parse_account_id(arn: &str) -> Result<AccountId, CredentialsError> {
```

Comment:
Maybe add a TODO here if we add a public ARN parser (which we should have an issue for, if not we should create one) we might want to replace this with that one day.

Reply:
Added `TODO` with an issue in fc1233e.